pkg/operation/common/extensions.go

```diff
@@ -125,7 +126,20 @@ func DeleteExtensionCR(
 		return err
 	}
 
-	return client.IgnoreNotFound(c.Delete(ctx, obj, deleteOpts...))
+	if err := client.IgnoreNotFound(c.Delete(ctx, obj, deleteOpts...)); err != nil {
+		return err
+	}
+
+	extensionStatus := obj.GetExtensionStatus()
+	if extensionStatus != nil {
+		for _, resource := range extensionStatus.GetResources() {
+			if err := utils.DeleteObjectByRef(ctx, c, &resource.ResourceRef, namespace); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
Hmm, I'm not sure about this - shouldn't the extension controller itself by responsible for cleaning up these referred resources on deletion as well as on migration?

Reply:
Removed this code as suggested.